node/src/actors/chain_manager/mining.rs

```diff
@@ -387,16 +389,41 @@ impl ChainManager {
                             &slice,
                             e
                         );
-                            None
+                            Some(
+                                RadonReport::from_result(Err(e), &ReportContext::default())
+                                    .unwrap(),
+                            )
                         },
                     );
 
-                    let rad_manager_addr = RadManager::from_registry();
-                    rad_manager_addr
-                        .send(RunTally {
-                            script: dr_state.data_request.data_request.tally.clone(),
-                            reveals: results.clone(),
-                        })
+                    let non_error_min =
+                        dr_state.data_request.min_consensus_percentage as f64 / 100.0;
+                    let len_results = results.len();
+                    let clause_result = tally_precondition_clause(results.clone(), non_error_min);
+
+                    let future_result = match clause_result {
+                        Ok((results, liars)) => {
+                            let rad_manager_addr = RadManager::from_registry();
+                            rad_manager_addr.send(RunTally {
+                                script: dr_state.data_request.data_request.tally.clone(),
+                                reveals: results,
+                                liars,
+                                error: None,
+                            })
+                        }
+
+                        Err(e) => {
+                            let rad_manager_addr = RadManager::from_registry();
+                            rad_manager_addr.send(RunTally {
+                                script: dr_state.data_request.data_request.tally.clone(),
+                                reveals: vec![],
```

Comment:
If `reveals` and `error` are exclusive, the API can be simplified to:

    reveals: Result<Vec<RadonTypes>, RadError>

```suggestion
                                reveals: Err(e),
```